### ISSUE TITLE
Global Hotkeys with DS4 Macros?

### ISSUE THREAD

**Issue by CapnCrinklepants** [Ryochan7/DS4Windows#343]
I have some OBS hotkeys bound to Ctrl+Shift+Space, and I'd like to create a macro for left analog click to trigger that macro. Works great! But only when OBS is the focused window. If I focus on any other screen, clicking the stick in doesn't work, but if I manually do Ctrl+Shift+Space, OBS does the thing. I can only imagine that DS4 funnels the input to the focused window as opposed to globally, but that seems a bit silly; especially in a case like this. Anyone encounter this and know how to change it? Is there a check box somewhere to let macros run globally that I'm missing? Is it a limitation of controllers at a deeper level? Thanks!

(Crossposted from reddit; this looks more active)

**Comment by Ryusennin** [comment]
Unless you created a specific profile for OBS, macros should work globally.

For example, I'm using a macro to capture screenshots via Radeon ReLive (Shift+Alt+S remapped to the Share button, PS4-style) and it works everywhere.

**Comment by Ryochan7** [comment]
Keyboard emulation is done globally using SendInput. It should not matter if OBS is the window currently in focus or not. I personally don't use the macro system in DS4Windows and I don't like it. It would not surprise me if DS4Windows did not hold all the keys down in a sequence while executing a macro. I will try that type of binding on my system and see what happens.

**Comment by sloanr333** [comment]
Just tested with Discord and I have the same issue there as well. The keybinds only work if I have either discord or the ds4 window focused... DInput mode only toggle doesn't make a difference either.

**Comment by Ryochan7** [comment]
I just got around to testing this; so punctual. I found that I had to add a wait in the macro in order for OBS to pick it up. A wait time had to be added between Space KeyDown and Space KeyUp. Otherwise, OBS would ignore it since keys would be pressed and released too quickly for the app to detect.

![macro_wait_example4](https://user-images.githubusercontent.com/581968/44304128-c08feb00-a318-11e8-911a-2a3c9e98913c.png)
